Client/Frontend/Widgets/PhotonActionSheetProtocol.swift

```diff
@@ -0,0 +1,200 @@
+    func getHomePanelActions(openURL: @escaping (URL) -> Void, vcDelegate: PageOptionsVC) -> [PhotonActionSheetItem] {
+        let openQR = PhotonActionSheetItem(title: "QR Scanner", iconString: "menu-ScanQRCode") { action in
+            let qrCodeViewController = QRCodeViewController()
+            qrCodeViewController.qrCodeDelegate = vcDelegate
+            let controller = UINavigationController(rootViewController: qrCodeViewController)
+            vcDelegate.present(controller, animated: true, completion: nil)
+        }
+        
+        let openSettings = PhotonActionSheetItem(title: "Settings", iconString: "menu-Settings") { action in
+            let settingsTableViewController = AppSettingsTableViewController()
+            settingsTableViewController.profile = self.profile
+            settingsTableViewController.tabManager = self.tabManager
+            settingsTableViewController.settingsDelegate = vcDelegate
+            
+            let controller = SettingsNavigationController(rootViewController: settingsTableViewController)
+            controller.popoverDelegate = vcDelegate
+            controller.modalPresentationStyle = UIModalPresentationStyle.formSheet
+            vcDelegate.present(controller, animated: true, completion: nil)
+        }
+        
+        let openTopSites = PhotonActionSheetItem(title: Strings.AppMenuTopSitesTitleString, iconString: "menu-panel-TopSites") { action in
+            openURL(HomePanelType.topSites.localhostURL)
+        }
+        
+        let openBookmarks = PhotonActionSheetItem(title: Strings.AppMenuBookmarksTitleString, iconString: "menu-panel-Bookmarks") { action in
+            openURL(HomePanelType.bookmarks.localhostURL)
+        }
+        
+        let openHistory = PhotonActionSheetItem(title: Strings.AppMenuHistoryTitleString, iconString: "menu-panel-History") { action in
+            openURL(HomePanelType.history.localhostURL)
+        }
+        
+        let openReadingList = PhotonActionSheetItem(title: Strings.AppMenuReadingListTitleString, iconString: "menu-panel-ReadingList") { action in
+            openURL(HomePanelType.readingList.localhostURL)
+        }
+        
+        return [openQR, openSettings, openTopSites, openBookmarks, openHistory, openReadingList]
+    }
+    
+    /*
+     Returns a list of actions which is used to build the general browser menu
+     These items repersent global options that are presented in the menu
+     TODO: These icons should all have the icons and use Strings.swift
+     */
+    
+    typealias PageOptionsVC = QRCodeViewControllerDelegate & SettingsDelegate & PresentingModalViewControllerDelegate & UIViewController
+    
+    func getOtherPanelActions() -> [PhotonActionSheetItem] {
+        
+        let noImageEnabled = NoImageModeHelper.isActivated(profile.prefs)
+        let noImageText = "Hide Images: \(noImageEnabled ? "On" : "Off")"
```

Comment:
Nit: we already have localized strings for this. 

Aside for less experienced readers: you should never concatenate localized strings.